lib/modules/showImages.js

```diff
@@ -472,11 +472,15 @@ addModule('showImages', (module, moduleID) => {
 	}
 
 	function updateRevealedImages(type) {
-		for (const image of imageList) {
-			if ($(image).hasClass(type) || image.imageLink.expandOnViewAll) {
-				revealImage(image, findImageFilter(image.imageLink));
-			}
+		let images = imageList.filter(v => $(v).hasClass(type) || v.imageLink.expandOnViewAll);
+
+		if (RESUtils.pageType() === 'linklist') {
+			images = images.map(v => [$(v).closest('.thing').data('rank'), v])
```

Comment:
> change the way imagelist is created and filled

That's a neat idea. How about something like .. in `findAllImages`, populate `imageList` with `ImagePromise` objects sorted by vertical position? This is easy to pull from `allElements` on pageload, but how how to splice in new ImagePromise when expandoing selftext or loading more comments? Should `imageList` be a tree structure mimicking the DOM tree?